### PR TITLE
Robot viewer

### PR DESCRIPTION
**Pull request by pensono**
Basic implementation

### CODE REVIEW

File: UI/Utilities/ObservableMap.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        public bool ContainsKey(string key) => dictionary.ContainsKey(key);
+
+        public T GetValue(string key) => dictionary[key].Value;
+
+        public void PutValue(string key, T value)
+        {
+            if (dictionary.ContainsKey(key))
+            {
+                dictionary[key].Value = value;
+            } else
```
Comment (Obbay2) [inline UI/Utilities/ObservableMap.cs]: Else should be on the next line :P

Comment (Obbay2) [inline UI/Utilities/ObservableMap.cs]: Else should be on the next line!

File: UI/Utilities/ObservableMap.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            {
+                dictionary[key] = new MeasuredValue<T>(value);
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs("Count"));
```
Comment (Obbay2) [inline UI/Utilities/ObservableMap.cs]: Is this "count" parameter going to be used for anything? If not is it needed?

Comment (pensono) [reply UI/Utilities/ObservableMap.cs]: Not that I'm aware of. I included it for completeness. Is it worth removing?

**Pull request merged**